lisa/sut_orchestrator/azure/platform_.py

```diff
@@ -521,34 +521,34 @@ def _prepare_environment(self, environment: Environment, log: Logger) -> bool:
         all_awaitable: bool = False
         errors: List[str] = []
 
-        for location in allowed_locations:
```

Comment:
This loop should check all locations already. Why move the loop into `_get_azure_capabilities`

Reply:
The loop was in `_prepare_environment` method. and `_get_azure_capabilities` method was returning an `awaitable_candidate` (if applicable) for the first location of the list. And even if the other locations have enough available quota, it was using the awaitable_candidate.

Reply:
The L538 `if all(isinstance(x, schema.NodeSpace) for x in caps):` is to check if all are available resources. If it's not, the loop takes the next location. Please check why it doesn't work as expected.

Reply:
IIUC, the check is to verify the object type. How does L538 verify if the quota is available or not? 
This https://github.com/microsoft/lisa/blob/main/lisa/sut_orchestrator/azure/platform_.py#L2592 could possibly return awaitable result.